backend/tests/test_email_parser.py

```
@@ -233,3 +233,10 @@ def test_parse_eml_extracts_reply_to_header():
         assert parsed["reply_to"] == "Reply Target <reply-target@test.com>"
     finally:
         os.unlink(temp_path)
+
+from unittest.mock import patch
+
+def test_parse_eml_mocked_oserror():
+    with patch("builtins.open", side_effect=OSError("Mocked OS Error")):
+        with pytest.raises(EmailParseError, match="Failed to read file dummy.eml: Mocked OS Error"):
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)